src/components/MovieListPage/MovieListPage.tsx

```diff
@@ -0,0 +1,235 @@
+import { MOVIE_MOCK } from '../../mocks/movie';
+import { GenreSelect } from '../GenreSelect/GenreSelect';
+import { MovieTile } from '../MovieTile/MovieTile';
+import { SortControl } from '../SortControl/SortControl';
+import {
+  MoviesContainer,
+  DetailsHeader,
+  DetailsContainer,
+  AddMovieButton,
+  Logo,
+  MovieListPageContainer,
+  MovieListPageFooter,
+  MoviesTotal,
+  SearchMovieContainer,
+  SearchMovieContent,
+  TopContainerHeader,
+  TopContainer,
+  MoviesGrid,
+  SearchSwitcherButton,
+  Icon,
+} from './MovieListPage.styled';
+import { Dialog, DialogProps } from '../Dialog/Dialog';
+import { Movie } from '../../types/movie';
+import { DialogContent, ConfirmButton } from '../Dialog/Dialog.styled';
+import { MovieDetails } from '../MovieDetails/MovieDetails';
+import { MovieForm } from '../MovieForm/MovieForm';
+import { SearchForm } from '../SearchForm/SearchForm';
+import { useEffect, useState } from 'react';
+import axios, { CancelTokenSource } from 'axios';
+import { MovieQueryParams, MoviesResponse } from '../../types/movies-response';
+import { MoviesService } from '../../services/movies.service';
+import { SORT_OPTIONS } from '../../constants/sort-options';
+import { useIsMount } from '../../hooks/useIsMount';
+
+const AppLogo = () => (
+  <Logo>
+    <span>netflix</span>
+    <span>roulette</span>
+  </Logo>
+);
+
+export function MovieListPage() {
+  const isMount = useIsMount();
+
+  const [cancelSource, setCancelSource] = useState<CancelTokenSource | null>(
+    null
+  );
+  const [showDialog, setShowDialog] = useState(false);
+  const [searchTerm, setSearchTerm] = useState('');
+  const [genre, setGenre] = useState('');
+  const [sort, setSort] = useState<keyof MoviesResponse>();
+  const [movieList, setMovieList] = useState<Movie[]>([]);
+  const [selectedMovie, setSelectedMovie] = useState<Movie>();
+  const [total, setTotal] = useState<number>();
+
+  const fetchData = async (params?: MovieQueryParams) => {
+    try {
+      if (cancelSource) {
+        cancelSource.cancel();
+        setCancelSource(null);
+      }
+      const source = axios.CancelToken.source();
+      setCancelSource(source);
+      const result = await new MoviesService().get(params, source);
+      setCancelSource(null);
+      setMovieList(result.data);
+      setTotal(result.totalAmount);
+    } catch (error) {
+      if (axios.isCancel(error)) return;
+    }
+  };
+
+  const [dialogContent, setDialogContent] = useState<DialogProps>({
+    children: null,
+    title: '',
+  });
+
+  useEffect(() => {
+    fetchData();
+  }, []);
+
+  useEffect(() => {
+    if (isMount) return;
+    fetchData({
+      search: searchTerm,
+      searchBy: 'title',
+      sortBy: sort,
+      sortOrder: 'asc',
+    });
+  }, [searchTerm]);
+
+  useEffect(() => {
+    if (isMount) return;
+    fetchData({
+      search: genre,
+      searchBy: 'genres',
+      sortBy: sort,
+      sortOrder: 'asc',
+    });
+  }, [genre]);
+
+  useEffect(() => {
```

Comment:
1) Please remove this piece from everywhere:
`if (isMount) return;` and `useIsMount` hook as well.

2) First three useEffects are redundant and leads to glitches in behaviour. 

3) Please remove them.

You can leave just this one useEffect, and pupulate dependency array with all required dependencies.

Please -dive deep in useEffect hook docks - it is super important to understand how this hook is working, when/how to use it and to keep track correct dependencies array .